.claude/positron-python.md

```diff
@@ -0,0 +1,151 @@
+
+## Project Structure
+```
+extensions/positron-python/
+├── src/                           # TypeScript extension code
+├── python_files/                  # Python runtime components
+│   ├── posit/                    # Positron-specific Python modules
+│   │   ├── positron/             # Core Positron Python functionality
+│   │   │   └── tests/            # Python unit tests for Positron modules
+│   │   └── pyproject.toml        # Python project configuration
+│   └── lib/                      # Vendored Python dependencies
+├── build/                        # Build configurations and CI helpers
+├── package.json                  # Node.js extension manifest
+└── requirements.txt              # Base Python dependencies
+```
+
+## Development Environment Setup
+
+We assume for now that the user is responsible for setting up their development environment.
+
+## Testing Framework
+
+### Python Testing
+All Python tests must be run from the correct directory with proper module resolution:
+
+```bash
+# Navigate to the Python module root (CRITICAL)
+cd extensions/positron-python/python_files/posit
+
+# Run all Positron Python tests
+python -m pytest
+
+# Run specific test modules
+python -m pytest positron/tests/test_data_explorer.py
+
+# Run with coverage
+python -m pytest --cov=positron --cov-report=term-missing
+
+# Run specific test with verbose output
+python -m pytest positron/tests/test_data_explorer.py::test_specific_function -v
+```
+
+### Test Requirements
+- **Module Resolution**: Always run pytest from `python_files/posit` directory
+- **No Main Blocks**: Never use `if __name__ == "__main__"` in test files
+
+## Code Quality & Linting
+
+### Automatic Formatting
+Code formatting is handled automatically via Claude Code hooks using ruff format.
+
+### Manual Quality Checks
+
+```bash
+# Python linting (from python_files/)
+ruff check .
+ruff format --check
+
+# Type checking with Pyright (from python_files/)
+pyright
+```
+
+### Tool Versions & Configuration
+- **Pyright**: Version 1.1.308 (matches CI)
+- **Ruff**: Line length 100 characters, target Python 3.8+
+- **Enabled rules**: Comprehensive set including flake8-bugbear, pycodestyle, isort, etc.
+- **Excludes**: Vendored dependencies (`lib/`, `posit/positron/_vendor/`)
+
+## Architecture & Key Components
+
+### Python Runtime Integration
+- **Kernel Management**: IPython kernel lifecycle and communication
+- **Environment Detection**: Virtual environments, conda, pyenv, pipenv, poetry
+- **Data Science Features**: Interactive data exploration and visualization
+- **Language Server**: Pylsp/Jedi integration for IntelliSense
+- **Debugging**: Integration with debugpy for Python debugging
+
```

Comment:
I feel like we could refine this but I'm not sure how specifically this section is trying to steer CC

Reply:
I'll just remove this generated section since it doesn't add much value